### PR TITLE
feat: support step-functions

### PR DESCRIPTION
**Pull request by dominhhai**
*Issue #, if available:*

*Description of changes:*

# Checklist

- [x] Tests have been added and are passing
- [x] Documentation has been updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by brettstack** [comment]
Thanks @dominhhai I'll take a look!

File: src/configure.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ function configure ({
       })
 
       try {
-        forwardRequestToNodeServer({
+        await forwardRequestToNodeServer({
```
Comment (brettstack) [inline src/configure.js]: Does this need to be async/await?

File: src/event-sources/aws/sqs.js [hunk 2]
```diff
@@ -1,18 +1,24 @@
 const getRequestValuesFromSqs = ({ event }) => {
-  const method = 'POST'
```
Comment (brettstack) [inline src/event-sources/aws/sqs.js]: Could you revert the style changes in this file?

File: src/event-sources/aws/sqs.js [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
-  }
-}
+    body,
+  };
+};
 
-const getResponseToSqs = ({ body }) => JSON.parse(body)
+const getResponseToSqs = ({ statusCode, body }) => {
+  if (statusCode >= 400) {
+    throw new Error(body);
```
Comment (brettstack) [inline src/event-sources/aws/sqs.js]: Could you explain this change please? I want to ensure you can still respond with `batchItemFailures`

File: src/event-sources/utils.js [hunk 4]
```diff
@@ -148,6 +152,17 @@ function getCommaDelimitedHeaders ({ headersMap, separator = ',', lowerCaseKey =
 
 const emptyResponseMapper = () => {}
 
+function emptyResponseMapperToEventSource({
+    statusCode,
+    body,
+}) {
+  if (statusCode >= 400) {
+    throw new Error(body)
```
Comment (brettstack) [inline src/event-sources/utils.js]: Could you explain this? Seems like it might be a breaking change

**Review by brettstack — CHANGES_REQUESTED** [review]
Looks good! Could you address comments? Tests and documentation would be awesome but I won't block on it 😀

File: src/configure.js (hunk 1 above)
Comment (dominhhai) [reply src/configure.js]: I tried to catch the exception from `forwardRequestToNodeServer`.

File: src/event-sources/aws/sqs.js (hunk 2 above)
Comment (dominhhai) [reply src/event-sources/aws/sqs.js]: I reverted these code

File: src/event-sources/aws/sqs.js (hunk 3 above)
Comment (dominhhai) [reply src/event-sources/aws/sqs.js]: When using HTTP server for handling the event source, the server may response the 5xx or 4xx http-code for any exception. This changing will useful for the case we using the same source code for handling http and event-sources.
In my case, I use Nestjs for handling event-source. The nestjs always handles the exception with 4xx http-code.

File: src/event-sources/utils.js (hunk 4 above)
Comment (dominhhai) [reply src/event-sources/utils.js]: It is same as the above
https://github.com/CodeGenieApp/serverless-express/pull/668#discussion_r1433596225

File: src/event-sources/aws/sqs.js (hunk 3 above)
Comment (brettstack) [reply src/event-sources/aws/sqs.js]: Let's revert this. I don't think we should throw errors at this level based on http code.

File: src/event-sources/utils.js (hunk 4 above)
Comment (brettstack) [reply src/event-sources/utils.js]: Let's revert this also.

File: src/event-sources/aws/sqs.js (hunk 3 above)
Comment (dominhhai) [reply src/event-sources/aws/sqs.js]: I reverted

File: src/event-sources/utils.js (hunk 4 above)
Comment (dominhhai) [reply src/event-sources/utils.js]: I reverted

File: src/configure.js (hunk 1 above)
Comment (dominhhai) [reply src/configure.js]: I reverted

**Review by brettstack — APPROVED** [review]

**Comment by brettstack** [comment]
Awesome work! I'll get this released.

**Comment by JarnoLeConte** [comment]
@dominhhai How to make use of this functionality? The `getEventSource` handler checks the following condition to determine if the source is `AWS_STEP_FUNCTIONS`:

```
if (event.context && event.context.Execution && event.context.State && event.context.StateMachine) {
  return 'AWS_STEP_FUNCTIONS'
}
```

Should I pass these context properties in "Step Functions" myself as parameters to the "Invoke Lambda" State. I would assume that this context properties should be passed to the handler automatically. I'm confused on how to make that work.

**Comment by dominhhai** [comment]
@JarnoLeConte 
Could you please describe your context?

**Pull request merged**